doc/spec/tags.md

```diff
@@ -0,0 +1,149 @@
+# Tags and Signatures
+
+## Introduction
+
+Classically, the role of tagging has been less formal in the docker eco-
+system. Effectively, the concept of "tagging" meant that one assigned a name
+to an image id.
+
+Tags in docker are missing several features that one might expect when coming
+from other systems. In git, one can tag any commit and then sign that tag,
+providing a name to a given revision and the ability to verify that the name
+was assigned by a trusted party. We'd like bring this 
+
+The following changes are proposed:
+
+1. Tags will become a first-class object in the image distribution data model.
+2. Tags are the primary mechanism for verification and named provenance.
+
+The tag object's role will be equivalent to the v2, schema version 1. Signatures 
+
+## Tags
+
+Tags are simply a named pointer to content. The content can be any blob but
+should mostly be a manifest. One can sign tags to later verify that they were
+created by a trusted party.
+
+This also relies on signatures being a first-class component. We may explore
+that in a separate proposal or integrate signatures completely with tags.
+
+### Routes
+
+The following routes are added to the V2 API specification:
+
+|Method|Path|Entity|Description|
+-------|----|------|------------
+| GET    | `/v2/<name>/tags/`         | Tag | List the tags for the repo. Must end with slash.      |
```

Comment:
If this is replacing an existing endpoint, how are the two expected to coexist? We ended up with the "tags/list" weirdness because of the variable length of <name>.